core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -788,7 +788,13 @@ private boolean addBlockBasedOnMinMaxValue(FilterExecuter filterExecuter, byte[]
       byte[][] minValue, boolean[] minMaxFlag, String filePath, int blockletId) {
     BitSet bitSet = null;
     if (filterExecuter instanceof ImplicitColumnFilterExecutor) {
-      String uniqueBlockPath = filePath.substring(filePath.lastIndexOf("/Part") + 1);
+      String uniqueBlockPath;
+      if (segmentPropertiesWrapper.getCarbonTable().isHivePartitionTable()) {
```

Comment:
In init method, blockletDataMapInfo already has carbonTable. Can use it, instead of getting from segmentPropertiesWrapper

Reply:
blockletDataMapInfo is not stored in BlockDataMap and we are getting it from DataMapModel.  In segmentPropertiesWrapper i just added getter for carbontable. i feel this is ok. Any suggestions ?

Reply:
Since you only need isHivePartitionTable and tablePath, can get from DataMapModel and assign to class-level variables

Reply:
For avoiding extra memory for two instance variables, we can use existing segmentPropertiesWrapper getter for carbontable.

Reply:
ok